diff --git a/android/build.gradle b/android/build.gradle
index fbd1d19..8a3536d 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -39,7 +39,7 @@ android {
       cmake {
         cppFlags "-O2 -frtti -fexceptions -Wall -fstack-protector-all"
         arguments "-DANDROID_STL=c++_shared", "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
-        abiFilters "x86", "x86_64", "armeabi-v7a", "arm64-v8a"
+        abiFilters(*rootProject.getProperties().get("reactNativeArchitectures").split(","))
       }
     }
   }
